app/views/products/show.html.haml

```diff
@@ -12,12 +12,16 @@
       = image_tag 'placeholder.jpg', class: "img-fluid card-img-top mh-100 mb-4"
   .col-6
     %h3
-      = @product.price
+      = number_to_currency(@product.price)
       $
```

Comment:
I'm not sure that you need dollar sign here, it should be shown as default unit https://apidock.com/rails/ActionView/Helpers/NumberHelper/number_to_currency